src/core/thread/mle.cpp

```diff
@@ -4469,5 +4487,22 @@ void Mle::UpdateAllDomainBackboneRouters(BackboneRouter::Leader::DomainPrefixSta
 }
 #endif // OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
 
+// Since Thread 1.2, MTD end device (including rx-on-when-idle or rx-off-when-idle) should
+// register or de-register their multicast address larger than realm local to the parent via
+// MLE Child Update Request.
+bool Mle::ShouldUpdateMulticastRegistration(void) const
```

Comment:
```suggestion
bool Mle::ShouldRegisterMulticastToParent(void) const
```
I think `ShouldUpdateMulticastRegistration` is not a good name because we can argue that FED sending `MLR.req` is also a way of updating Multicast Registration?

Reply:
renamed.